cobalt/browser/h5vcc_runtime/deep_link_manager.cc

```diff
@@ -49,5 +49,28 @@ const std::string DeepLinkManager::GetAndClearDeepLink() {
   return value;
 }
 
+void DeepLinkManager::AddListener(
+    mojo::Remote<DeepLinkListener> listener_remote) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  // mojo::RemoteSet removes the listener_remote automatically when the Mojo
+  // client disconnects
+  listeners_.Add(std::move(listener_remote));
+}
+
+void DeepLinkManager::OnDeepLink(const std::string& url) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  if (listeners_.empty()) {
+    // Deeplink is held until a callback is registered, at which point it will
+    // be consumed."
```

Comment:
QQ: Is this so? If yes then I'd expect l.57 to also do sth like:
```
  if (deep_link_) {
    listener_remote->ProcessDeepLink(url);
  }
  listeners_.Add(std::move(listener_remote));
```

But I thought that would not be the case: a remote listener would not be warm deep links received in DeepLinkManager before its registration.